DXMainClient/DXGUI/Multiplayer/GameLobby/GameLobbyBase.cs

```diff
@@ -1520,11 +1521,96 @@ private void WriteMap(PlayerHouseInfo[] houseInfos)
+        {
+            var mapFileInfo = new FileInfo(mapIni.FileName);
+            string mapFileBaseName = Path.GetFileNameWithoutExtension(mapFileInfo.Name);
+
+            IEnumerable<string> supplementalMapFiles = GetSupplementalMapFiles(mapFileInfo.DirectoryName, mapFileBaseName).ToList();
+            if (!supplementalMapFiles.Any())
+                return;
+
+            List<string> supplementalFileNames = new();
+            foreach (string file in supplementalMapFiles)
+            {
+                try
+                {
+                    // Copy each supplemental file
+                    string supplementalFileName = $"spawnmap{Path.GetExtension(file)}";
+                    File.Copy(file, SafePath.CombineFilePath(ProgramConstants.GamePath, supplementalFileName), true);
+                    supplementalFileNames.Add(supplementalFileName);
+                }
+                catch (Exception e)
+                {
+                    string errorMessage = "Unable to copy supplemental map file".L10N("Client:Main:SupplementalFileCopyError") + $" {file}";
+                    Logger.Log(errorMessage);
+                    Logger.Log(e.Message);
+                    XNAMessageBox.Show(WindowManager, "Error".L10N("Client:Main:Error"), errorMessage);
+                    
+                }
+            }
+            
+            // Write the supplemental map files to the INI (eventual spawnmap.ini)
+            mapIni.SetStringValue("Basic", "SupplementalFiles", string.Join(",", supplementalFileNames));
+        }
+
+        /// <summary>
+        /// Delete all supplemental map files from last spawn
+        /// </summary>
+        private void DeleteSupplementalMapFiles()
+        {
+            IEnumerable<string> supplementalMapFilePaths = GetSupplementalMapFiles(ProgramConstants.GamePath, "spawnmap").ToList();
+            if (!supplementalMapFilePaths.Any())
+                return;
+
+            foreach (string supplementalMapFilename in supplementalMapFilePaths)
+            {
+                try
+                {
+                    File.Delete(supplementalMapFilename);
+                }
+                catch (Exception e)
+                {
+                    string errorMessage = "Unable to delete supplemental map file".L10N("Client:Main:SupplementalFileDeleteError") + $" {supplementalMapFilename}";
+                    Logger.Log(errorMessage);
+                    Logger.Log(e.Message);
+                    XNAMessageBox.Show(WindowManager, "Error".L10N("Client:Main:Error"), errorMessage);
+                }
+            }
+        }
+
+        private static IEnumerable<string> GetSupplementalMapFiles(string basePath, string baseFileName)
+        {
+            // Get the supplemental file names for allowable extensions
+            var supplementalMapFileNames = ClientConfiguration.Instance.SupplementalMapFileExtensions
+                .Select(ext => $"{baseFileName}.{ext}".ToLower())
```

Comment:
```suggestion
                .Select(ext => $"{baseFileName}.{ext}".ToUpperInvariant())
```

Reply:
Check out [Globalization rules](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/globalization-warnings) for more info.